src/main/java/pl/lodz/p/it/zzpj/botsite/web/controllers/UserController.java

```diff
@@ -30,9 +41,35 @@ public UserController(ModelMapper modelMapper, UserService userService) {
             produces = MediaType.APPLICATION_JSON_VALUE
     )
     @ResponseStatus(HttpStatus.OK)
-    public UserRegistrationDto registerUser(@RequestBody UserRegistrationDto dto) throws UsernameAlreadyExistsException, UserAdditionException {
+    public UserRegistrationDto registerUser(@RequestBody UserRegistrationDto dto)
+            throws UserAdditionException, UsernameAlreadyExistsException {
         User user = this.modelMapper.map(dto, User.class);
         User userRegistered = this.userService.addUser(user);
+
+        this.eventPublisher.publishEvent(new OnUserRegistrationCompleteEvent(user));
```

Comment:
przesunąłbym taką logikę niżej w hierarchii i ukrył za wrappowanym obiektem z interfejsem, dużo bardziej elastyczne i pozwoli *wyłączyć* maila jedną zmianą gdy to będzie konieczne.